kartothek/serialization/_parquet.py

```diff
@@ -445,10 +447,18 @@ def _predicate_accepts(predicate, row_meta, arrow_schema, parquet_reader):
     if isinstance(val, float):
         min_value -= _epsilon(min_value)
         max_value += _epsilon(max_value)
+
+    # TODO: What to do for "<=", "<", ">", ">=" np.nan?
```

Comment:
We should have a "verify valid predicate" function somewhere. I would implement this check there. Nulls are not a valid input for these operators.

Reply:
It's called `check_predicates` in `_generic.py`